Formatting jobs fails in CI

I'm not touching anything in the runtime and still the formatting job fails: https://dev.azure.com/dnceng/public/_build/results?buildId=1287580&view=logs&jobId=db5b9618-eefc-58e8-4554-fbce452d4dac&j=db5b9618-eefc-58e8-4554-fbce452d4dac&t=29bb2718-3138-535b-c6ac-1a4597023815

```
Downloading https://raw.githubusercontent.com/dotnet/jitutils/main/bootstrap.cmd to C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\bootstrap.cmd
Convert C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\bootstrap.cmd to Windows line endings
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\bootstrap.cmd
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Checked -o windows -c D:\a\1\s\src\coreclr --verbose --projects dll
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Checked -o windows -c D:\a\1\s\src\coreclr --verbose --projects standalone
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Checked -o windows -c D:\a\1\s\src\coreclr --verbose --projects crossgen
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Debug -o windows -c D:\a\1\s\src\coreclr --verbose --projects dll
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Debug -o windows -c D:\a\1\s\src\coreclr --verbose --projects standalone
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Debug -o windows -c D:\a\1\s\src\coreclr --verbose --projects crossgen
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Release -o windows -c D:\a\1\s\src\coreclr --verbose --projects dll
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Release -o windows -c D:\a\1\s\src\coreclr --verbose --projects standalone
Running: C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\jitutils\bin\jit-format.bat -a x64 -b Release -o windows -c D:\a\1\s\src\coreclr --verbose --projects crossgen
Creating patch file D:\a\1\s\src\coreclr\format.patch
Deleting C:\Users\VSSADM~1\AppData\Local\Temp\tmp9r8m9v72\bootstrap.cmd
There were errors in formatting. Please run jit-format locally with: 

	jit-format -a x64 -b Checked -o windows -c <absolute-path-to-coreclr> --verbose --fix --projects dll


Or download and apply generated patch:
1. From the GitHub 'Checks' page on the Pull Request, with the failing Formatting
   job selected (e.g., 'Formatting Linux x64'), click the 'View more details on
   Azure Pipelines' link.
2. Select the '1 artifact produced' at the end of the log.
3. Artifacts are located in alphabetical order, target artifact name is
   'format.<OS>.<architecture>.patch.'. Find appropriate format patch artifact.
4. On the right side of the artifact there is a 'More actions' menu shown by a
   vertical three-dot symbol. Click on it and select 'Download artifacts' option.
5. Unzip the patch file.
6. git apply format.patch
##[error](NETCORE_ENGINEERING_TELEMETRY=Build) Format job found errors, please apply the format patch.
##[error]The process 'C:\hostedtoolcache\windows\Python\3.9.6\x64\python.exe' failed with exit code 4294967295
```

--- a/eng/formatting/download-tools.ps1
+++ b/eng/formatting/download-tools.ps1
@@ -1,3 +1,4 @@
+. "$PSScriptRoot/../common/tools.ps1" # for Retry function
 
 function DownloadClangTool {
     param (
@@ -10,29 +11,13 @@ function DownloadClangTool {
     $baseUri = "https://clrjit.blob.core.windows.net/clang-tools/windows"
 
     if (-not $(ls $downloadOutputPath | Where-Object { $_.Name -eq "$toolName.exe" })) {
-        $baseBackoffSeconds = 2;
 
-        $success = $false
-        for ($i = 0; $i -lt 5; $i++) {
-            Write-Output "Attempting download of '$baseUri/$toolName.exe'"
-            try {
-                # Pass -PassThru as otherwise Invoke-WebRequest leaves a corrupted file if the download fails. With -PassThru the download is buffered first.
-                # -UseBasicParsing is necessary for older PowerShells when Internet Explorer might not be installed/configured
-                $null = Invoke-WebRequest -Uri "$baseUri/$toolName.exe" -OutFile $(Join-Path $downloadOutputPath -ChildPath "$toolName.exe") -PassThru -UseBasicParsing
-                $success = $true
-                break
-            }
-            catch {
-                Write-Output $_
-            }
-			
-            Write-Output "Download attempt $($i+1) failed. Trying again in $($baseBackoffSeconds + $baseBackoffSeconds * $i) seconds"
-            Start-Sleep -Seconds $($baseBackoffSeconds + $baseBackoffSeconds * $i)
-        }
-        if (-not $success) {
-            Write-Output "Failed to download $toolName"
-            throw [System.IO.IOException] "Could not download $toolName"
-        }
+        Retry({
+            Write-Output "Downloading '$baseUri/$toolName.exe'"
+            # Pass -PassThru as otherwise Invoke-WebRequest leaves a corrupted file if the download fails. With -PassThru the download is buffered first.
+            # -UseBasicParsing is necessary for older PowerShells when Internet Explorer might not be installed/configured
+            $null = Invoke-WebRequest -Uri "$baseUri/$toolName.exe" -OutFile $(Join-Path $downloadOutputPath -ChildPath "$toolName.exe") -PassThru -UseBasicParsing
+        })
     }
 }
 

--- a/eng/formatting/download-tools.ps1
+++ b/eng/formatting/download-tools.ps1
@@ -17,7 +17,8 @@ function DownloadClangTool {
             Write-Output "Attempting download of '$baseUri/$toolName.exe'"
             try {
                 # Pass -PassThru as otherwise Invoke-WebRequest leaves a corrupted file if the download fails. With -PassThru the download is buffered first.
-                $null = Invoke-WebRequest -Uri "$baseUri/$toolName.exe" -OutFile $(Join-Path $downloadOutputPath -ChildPath "$toolName.exe") -PassThru
+                # -UseBasicParsing is necessary for older PowerShells when Internet Explorer might not be installed/configured
+                $null = Invoke-WebRequest -Uri "$baseUri/$toolName.exe" -OutFile $(Join-Path $downloadOutputPath -ChildPath "$toolName.exe") -PassThru -UseBasicParsing
                 $success = $true
                 break
             }

--- a/eng/formatting/download-tools.ps1
+++ b/eng/formatting/download-tools.ps1
@@ -1,6 +1,5 @@
 
-function DownloadClangTool
-{
+function DownloadClangTool {
     param (
         [string]
         $toolName,
@@ -10,27 +9,28 @@ function DownloadClangTool
 
     $baseUri = "https://clrjit.blob.core.windows.net/clang-tools/windows"
 
-    if (-not $(ls $downloadOutputPath | Where-Object {$_.Name -eq "$toolName.exe"}))
-    {
+    if (-not $(ls $downloadOutputPath | Where-Object { $_.Name -eq "$toolName.exe" })) {
         $baseBackoffSeconds = 2;
 
         $success = $false
         for ($i = 0; $i -lt 5; $i++) {
-            echo "Attempting download of '$baseUri/$toolName.exe'"
-            $status = Invoke-WebRequest -Uri "$baseUri/$toolName.exe" -OutFile $(Join-Path $downloadOutputPath -ChildPath "$toolName.exe")
-            if ($status.StatusCode -lt 400)
-            {
+            Write-Output "Attempting download of '$baseUri/$toolName.exe'"
+            try {
+                # Pass -PassThru as otherwise Invoke-WebRequest leaves a corrupted file if the download fails. With -PassThru the download is buffered first.
+                $null = Invoke-WebRequest -Uri "$baseUri/$toolName.exe" -OutFile $(Join-Path $downloadOutputPath -ChildPath "$toolName.exe") -PassThru
                 $success = $true
                 break
-            } else {
-                echo "Download attempt $($i+1) failed. Trying again in $($baseBackoffSeconds + $baseBackoffSeconds * $i) seconds"
-                Start-Sleep -Seconds $($baseBackoffSeconds + $baseBackoffSeconds * $i)
             }
+            catch {
+                Write-Output $_
+            }
+			
+            Write-Output "Download attempt $($i+1) failed. Trying again in $($baseBackoffSeconds + $baseBackoffSeconds * $i) seconds"
+            Start-Sleep -Seconds $($baseBackoffSeconds + $baseBackoffSeconds * $i)
         }
-        if (-not $success)
-        {
-            Write-Output "Failed to download clang-format"
-            return 1
+        if (-not $success) {
+            Write-Output "Failed to download $toolName"
+            throw [System.IO.IOException] "Could not download $toolName"
         }
     }
 }

